src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/BaseRunner.cs

```diff
@@ -231,48 +233,74 @@ private async Task<(TestResult, string)> RunAsync2(TestCase testCase)
                         return (TestResult.Pass, null);
                     }
 
-                    var actualErrorKind = errorResult.Errors.First().Kind;
-                    if (int.TryParse(expectedErrorKind, out var numericErrorKind))
+                    var expectedErrorKinds = expectedErrorKind.Split(',', StringSplitOptions.RemoveEmptyEntries).Select(x => x.Trim()).ToArray();
+
+                    if (errorResult.Errors.Count != expectedErrorKinds.Count())
                     {
-                        // Error given as the internal value
-                        if (numericErrorKind == (int)actualErrorKind)
-                        {
-                            return (TestResult.Pass, null);
-                        }
-                        else
-                        {
-                            return (TestResult.Fail, $"Received an error, but expected kind={expectedErrorKind} and received {actualErrorKind} ({(int)actualErrorKind})");
-                        }
+                        return (TestResult.Fail, $"Received {errorResult.Errors.Count} errors while we were expecting {expectedErrorKinds.Count()} errors.\r\n" +
+                                                 $"Received {string.Join(", ", errorResult.Errors.Select(e => e.Kind))} and was expecting {expectedErrorKind}");
                     }
-                    else if (Enum.TryParse<ErrorKind>(expectedErrorKind, out var errorKind))
+
+                    var results = errorResult.Errors.Select((er, i) =>
                     {
-                        // Error given as the enum name
-                        if (errorKind == actualErrorKind)
+                        var actualErrorKind = er.Kind;
+                        var eek = expectedErrorKinds[i];
```

Comment:
>eek [](http://example.com/codeflow?start=28&length=3)

[minor] rename to `expectedErrorKinds` or `expectedKinds` - easier to read

Reply:
done